client/extensions/wp-super-cache/state/data-layer/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+  WP_SUPER_CACHE_TEST_CACHE,
+  WP_SUPER_CACHE_RECEIVE_TEST_CACHE_RESULTS,
+} from '../action-types';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { addHandlers } from 'state/data-layer/extensions-middleware';
+
+const testCache = ( { dispatch }, action ) => {
+	const { siteId, httpOnly } = action;
+	dispatch( http( {
+		method: 'POST',
+		apiVersion: '1.2',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			path: '/wp-super-cache/v1/cache/test',
+			body: JSON.stringify( { httponly: httpOnly } ),
+			json: true
+		}
+	}, action ) );
+};
+
+const receiveSuccess = ( { dispatch }, { siteId }, next, data ) => {
+	dispatch( { type: WP_SUPER_CACHE_RECEIVE_TEST_CACHE_RESULTS, siteId, results: data } );
+	dispatch( successNotice( i18n.translate( 'Cache test completed successfully.' ) ) );
+};
```

Comment:
another no-big-deal is the naming here which can carry more semantic meaning if we want. instead of `receiveSuccess` we could say something like `addResultsAndAnnounce` (fewer `_RECEIVE` actions and more `_ADD` actions) while below it could be something like `announceFailure`